pkg/podman/podman.go

```diff
@@ -0,0 +1,188 @@
+package podman
+
+import (
+	"archive/tar"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/buildah/define"
+	"github.com/containers/podman/v4/pkg/bindings"
+	"github.com/containers/podman/v4/pkg/bindings/containers"
+	"github.com/containers/podman/v4/pkg/bindings/images"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/containers/podman/v4/pkg/specgen"
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"go.uber.org/zap"
+)
+
+const (
+	podmanSock         = "unix:///run/podman/podman.sock"
+	dockerfile         = "Dockerfile"
+	podmanDirName      = "podman"
+	podmanBuildLogFile = "podman-image-build.log"
+)
+
+type Podman struct {
+	context context.Context
+	socket  string
+	out     string
+}
+
+// New setups a podman listening service and returns a connected podman client.
+//
+// Parameters:
+//   - out - location for podman to output any logs created as a result of podman commands
+func New(out string) (*Podman, error) {
+	podmanDirPath := filepath.Join(out, podmanDirName)
```

Comment:
I'm not entirely sure if we want this. After the recent build directory restructuring all log files are now stored at the root `BuildDir`. This makes it very easy to follow and I'm afraid if we start introducing subdirectories only for the sake of log files then we will make troubleshooting more confusing. After all, everything would also need to be covered by tests and docs. What are your thoughts?

Reply:
Agreed, having all log files in a single place would make it easier for the user to debug possible problems. Also the user would not have to worry if he missed any log files.